lib/exchanges.js

```diff
@@ -0,0 +1,94 @@
+var base      = require('taskcluster-base');
+var assert    = require('assert');
+var _         = require('lodash');
+
+// Common schema prefix
+var SCHEMA_PREFIX_CONST = 'http://schemas.taskcluster.net/github/v1/';
+
+/** Build common routing key construct for `exchanges.declare` */
+var commonRoutingKey = function(options) {
+    options = options || {};
+     let routingKey = [
+        {
+          name:             'routingKeyKind',
+          summary:          "Identifier for the routing-key kind. This is " +
+                            "always `'primary'` for the formalized routing key.",
+          constant:         'primary',
+          required:         true
+        }, {
+          name:             'organization',
+          summary:          "The GitHub `organization` which had an event.",
+          maxSize:          100,
+          required:         true
+        }, {
+          name:             'repository',
+          summary:          "The GitHub `repository` which had an event.",
```

Comment:
random note, if github had allowed very long names, such that they wouldn't fit in the routing key... We could have hashed them and taken the first 12 characters or so.. Collision would be unlikely... That's a pattern for me to remember :)